grinder/core.py

```diff
@@ -129,6 +131,32 @@ def __init__(
         self.filemanager = GrinderFileManager()
         self.db = GrinderDatabase()
 
+    def onehost(self, host_address: str) -> List[dict]:
+
+        if self.shodan_api_key == "YOUR_DEFAULT_API_KEY":
+            print(f"│ Shodan key is not defined. Skip scan.")
+            print(f"└ ", end="")
+            return []
+
+        dnsResolve = 'https://api.shodan.io/dns/resolve?hostnames=' + host_address + '&key=' + self.shodan_api_key
```

Comment:
Why are you using Shodan here? We can resolve DNS from the framework itself, with something like `host`, `whois`, any other utility, or maybe something even lighter.

Reply:
Never use plus sign for string concatenation. Use format strings.

Reply:
Naming convention - it's not JavaScript. Use underscore.